sotodlib/tod_ops/binning.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import logging
+logger = logging.getLogger(__name__)
+
+def bin_signal(aman, bin_by, signal=None,
+               bin_range=None, bins=100, flags=None):
+    """
+    Bin time-ordered data by the `bin_by` and return the binned signal and its standard deviation.
+
+    Parameters
+    ----------
+    aman : TOD
+        The Axismanager object to be binned.
+    bin_by : array-like
+        The array by which signal is binned. It should has the same samps as aman.
+    signal : str, optional
+        The name of the signal to be binned. Defaults to aman.signal if not specified.
+    bin_range : list or None
+        A list specifying the bin range ([min, max])
+        Default is None, which means bin range is set to [min(bin_by), max(bin_by)] automaticaly by np.histogram.
+    bins : int, optional
+        The number of bins to use. Default is 100.
```

Comment:
In other binning functions, such as numpy.histogram, "bins" can be a number of bins (like this) or an array with the bin edges.

It would be nice to imitate that interface -- if bins is passed in as an array, it is used as the bins.